sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentServiceRequest.java

```diff
@@ -953,7 +953,7 @@ public void clearRoutingHints() {
             return Flux.empty();
         }
 
-        return Flux.just(Unpooled.wrappedBuffer(contentAsByteArray));
+        return Flux.just(Unpooled.wrappedBuffer(contentAsByteArray).retain());
```

Comment:
seems you are trying to fix a netty ref counting issue.

I don't have all the context. However at the first look this doesn't seem to me to be the right place for increasing the ref count.

`Unpooled.wrappedBuffer()` already sets ref count to a non zero value. If we have ref counting issue. I suspect we should fix that elsewhere. not here.

Reply:
This is done, resolving the conversation.